07_boolean_2/main.go

```diff
@@ -15,56 +15,262 @@ E.g.:
+	// NAND // true, true == false || true, false == true, || false, false == true || false, true == true
+	fmt.Println("NAND", !bool1 || !bool2)
+	// Alternative
+	fmt.Println("NAND", !(bool1 && bool2))
+	//  NOR // true, true ==  false || true, false == false || false, false == true || false, true == false
+	fmt.Println("NOR", !(bool1 || bool2))
+	// // XOR // true, true == false || true, false == true || false, false == false || false, true == true
+	fmt.Println("XOR", (bool1 || bool2) && (!bool1 || !bool2))
+}
+
+// TODO: Write a function that takes a string and an integer and performs
+// the following checks:
+// - If the string is empty, return "empty"
+// - If the string is longer than the value of the integer
+//   - If the integer is even, return "longer and even"
+//   - If the integer is odd, return "longer and odd"
+//
+// - If the string is shorter than the integer, return "shorter"
+func strIntCheck(str string, num int) {
+	// var stringLength = len(str)
+	// var integerLength = len(num)
+	if str == "" {
+		fmt.Println("empty")
+	} else if len(str) > num {
+		if len(str)%2 == 0 {
+			fmt.Println("longer and even")
+		} else {
+			fmt.Println("longer and odd")
+		}
+	} else {
+		fmt.Println("shorter")
+	}
+}
+
+// TODO: Write a function that takes 5 integers (grades) and returns the
+// following:
+// - If any of the grades is less than 0 or greater than 100, return
+// "invalid grade".
+// - If the average of the grades is greater than or equal to 90
+//   - If no grade is below 80, return "A+"
+//   - If no grade is below 70, return "A"
+//   - If any grade is below 70, return "A-"
+//
+// - If the average of the grades is greater than or equal to 80, return
+// "B"
+//   - If no grade is below 80, return "B+"
+//   - If no grade is below 70, return "B"
+//   - If any grade is below 60, return "B-"
+//
+// - If the average of the grades is greater than or equal to 70, return
+// "C"
+//
+//   - If no grade is below 70, return "C+"
+//
+//   - If no grade is below 60, return "C"
+//
+//   - If any grade is below 60, return "C-"
+
+// if grade.. > 100 || grade.. < 0 return invalid
+func gradeValid(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int) bool {
+	if (grade1 < 0 || grade1 > 100) || (grade2 < 0 || grade2 > 100) || (grade3 < 0 || grade3 > 100) || (grade4 < 0 || grade4 > 100) || (grade5 < 0 || grade5 > 100) {
+		return false
+	} else {
+		return true
+	}
+}
+
+// check if any grade is < x
+func anyGradeUnder(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int, limit int) bool {
+	if grade1 < limit || grade2 < limit || grade3 < limit || grade4 < limit || grade5 < limit {
+		return true
+	} else {
+		return false
+	}
+}
+
+// check if all grades are >= x
+func allGradesOver(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int, limit int) bool {
+	if grade1 >= limit && grade2 >= limit && grade3 >= limit && grade4 >= limit && grade5 >= limit {
+		return true
+	} else {
+		return false
+	}
+}
```

Comment:
Similar idea here, try to do an early return pattern.

Reply:
Done